azavea/pfb-network-connectivity
# Make max trip distance configurable by UI (#866)

## Overview

Client wanted to run interactive and batch analyses using
the angularjs client without having to use the default max
trip distance value of 2680 meters. This was previously only
possible by setting the NB_MAX_TRIP_DISTANCE environment
variable before running `scripts/run-local-analysis` locally.

### Demo

![Screenshot from 2022-04-01 15-59-17](https://user-images.githubusercontent.com/36001800/161333203-c3c6afc6-7006-4af4-baed-611e458439a7.png)
![Screenshot from 2022-04-01 15-59-25](https://user-images.githubusercontent.com/36001800/161333205-06006468-f21d-43f7-8283-beaf483419ac.png)


## Testing Instructions

 * `./scripts/setup`
 * in another tab `./scripts/server`
 * go to http://localhost:9301/admin#/admin/analysis-jobs/create/batch/ 
 * give it a shapefile (feel free to slack me for one) and a trip distance
 * check that the logged commands have the NB_MAX_TRIP_DISTANCE you set
 * check that the analysis jobs in the UI have the correct max trip distance
 * go to http://localhost:9301/admin#/admin/analysis-jobs/create/
 * pick a neighborhood and set a different trip distance from before
 * check that the logged command has the new NB_MAX_TRIP_DISTANCE you set
 *  check that the analysis job in the UI has the new correct max trip distance


## Checklist


Resolves #866
